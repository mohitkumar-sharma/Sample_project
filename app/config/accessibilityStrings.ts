export default {
  NORMAL_BUTTON_TEXT: 'normal-button-text',
  SLIDER_BUTTON_TEXT: 'slider-button-text',
  NORMAL_BUTTON: 'normal-button',
  SLIDER_BUTTON_ICON: 'slider-button-icon',
  SLIDER_BUTTON_CONTAINER: 'slider-button-container',
  BANNER_TEXT: 'banner-text',
  HEADER_SIDES_TEXT: 'header-sides-text',
  HEADER_PRESSABLE_ELEMENT: 'header-pressable-element',
  SETTINGS_SWITCH_TEXT: 'settings-switch-text',
  SETTINGS_SWITCH: 'settings-switch',
  INV_TEXT_INPUT: 'inv-text-input',
};
