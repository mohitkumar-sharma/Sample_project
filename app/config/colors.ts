export default {
  black: '#000000',
  gray: '#9B9B9B',
  divider: '#E8E8E8',
  white: '#FFFFFF',
  themeBlue: '#6EB1F7',
  themeLightBlack: '#34424A',
  appTheme: '#1A1924',
  borderLine: '#59606A',
  warning: '#FFCC00',
  red: '#FF0000',
};
