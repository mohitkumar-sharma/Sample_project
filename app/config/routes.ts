export default {
  WELCOME: 'Welcome',
  HOME: 'Home',
  SETTINGS: 'Settings',
};
