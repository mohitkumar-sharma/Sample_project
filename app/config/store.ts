import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { rootReducer } from '../states';

const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware(),
});

export type AppDispatch = typeof store.dispatch;

export { store };
