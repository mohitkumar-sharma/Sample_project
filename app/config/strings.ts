export default {
  GO_TO_HOME: 'Go To Home',
  DIAMOND_ICON: 'diamond',
  SLIDE_ME: 'Slide me to continue',
  APP_RUNNING_WARNING: 'Notice: You are running this app on',
  WELCOME_INPUT_PLACEHOLDER: 'Enter your name here...',
  SIGN_OUT: 'SignOut',
  HOME_BUTTONS_HEADING: '4 variations of a button',
  PRESS_ME: 'Press me',
  TRANSPARENT_BTN_PRESSED: 'Transparent button is pressed',
  LIGHT_BLACK_BTN_PRESSED: 'Light black button is pressed',
  BLUE_BTN_PRESSED: 'Blue button is pressed',
  SWITCH_TEXT: 'Enable Warning Banner',
  ALERT_TITLE_INVALID_USER_NAME: 'Invalid UserName!',
  ALERT_TITLE_BTN_PRESSED: 'Button Pressed!',
  ALERT_ENTER_VALID_USER_NAME: 'Please enter a valid user name.',
};
