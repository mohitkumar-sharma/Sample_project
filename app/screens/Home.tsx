import React, { useEffect, useCallback } from 'react';
import { View, Text, StyleSheet, SafeAreaView, Alert } from 'react-native';
import config from '../config';
import { INVButton, INVSliderButton, HeaderPressableText, INVBanner } from '../components';
import { useDispatch, useSelector } from 'react-redux';
import { UserInfo, DeviceInfo, SettingsInfo } from '../states';
import { StackNavigationProp } from '@react-navigation/stack';
import { MainStackParamList } from '../navigation/Main';
import { useBackButton } from '../hooks';

/* Props for navigation*/
type NavigationProps = {
  navigation: StackNavigationProp<MainStackParamList, 'Home'>;
};

const Home: React.FC<NavigationProps> = ({ navigation }): JSX.Element => {
  /* useDispatch hook from react-redux for dispatching actions*/
  const dispatch = useDispatch();

  /* useSelector hook from react-redux to select device type from redux */
  const deviceType = useSelector(DeviceInfo.selectDeviceType);
  /* useSelector hook from react-redux to select warning banner from redux */
  const isShowBanner = useSelector(SettingsInfo.selectWarningBannerStatus);

  /**
   * This function is responsible to handle the hardware button of android
   */
  const backButtonHandler = () => {
    navigation.goBack();
    return true;
  };

  /* custom hook called from hook directory */
  useBackButton(backButtonHandler);

  /**
   * This function is responsible to clear the user name stored in redux
   */
  const onSignOutPress = useCallback(() => {
    dispatch(UserInfo.saveUser({ userName: '' }));
    dispatch(SettingsInfo.saveWarningBannerStatus({ isShowWarningBanner: true }));
  }, [dispatch]);

  /**
   * This function is responsible for setting a signout button on left side of header
   */
  const setNavigationOptions = useCallback(() => {
    navigation.setOptions({
      headerLeft: () => (
        <HeaderPressableText
          text={config.strings.SIGN_OUT}
          containerStyle={styles.headerLeftContainer}
          textStyle={styles.headerLeftText}
          onPress={onSignOutPress}
        />
      ),
    });
  }, [navigation, onSignOutPress]);

  /* useEffect hook to set the navigation options */
  useEffect(() => {
    setNavigationOptions();
  }, [setNavigationOptions]);

  /**
   * This function is called when a button pressed
   */
  const buttonOnPress = (btnString: string) => {
    Alert.alert(config.strings.ALERT_TITLE_BTN_PRESSED, btnString);
  };

  return (
    <SafeAreaView style={styles.container}>
      {isShowBanner && deviceType !== 'Device' ? (
        <INVBanner text={`${config.strings.APP_RUNNING_WARNING} ${deviceType}`} />
      ) : null}
      <View style={styles.wrapper}>
        <Text style={styles.text}>{config.strings.HOME_BUTTONS_HEADING}</Text>
        <INVButton
          containerStyle={styles.firstBtnContainer}
          buttonText={config.strings.PRESS_ME}
          onPress={() => buttonOnPress(config.strings.TRANSPARENT_BTN_PRESSED)}
          buttonTextStyle={styles.commonBtnText}
        />
        <INVButton
          containerStyle={styles.secondBtnContainer}
          buttonText={config.strings.PRESS_ME}
          onPress={() => buttonOnPress(config.strings.LIGHT_BLACK_BTN_PRESSED)}
          buttonTextStyle={styles.commonBtnText}
        />
        <INVButton
          containerStyle={styles.thirdBtnContainer}
          buttonText={config.strings.PRESS_ME}
          onPress={() => buttonOnPress(config.strings.BLUE_BTN_PRESSED)}
        />
        <INVSliderButton
          buttonText={config.strings.SLIDE_ME}
          buttonTextStyle={styles.commonBtnText}
          containerStyle={styles.sliderBtnContainer}
          iconName={config.strings.DIAMOND_ICON}
          onEndReached={() => navigation.navigate('Settings')}
        />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: config.colors.appTheme },
  wrapper: { bottom: 20, position: 'absolute', width: '100%' },
  text: {
    textAlign: 'center',
    fontSize: 18,
    color: config.colors.warning,
    fontWeight: config.fontWeights.FONT_WEIGHT_MEDIUM,
    padding: 5,
  },
  firstBtnContainer: { marginTop: 15, backgroundColor: 'transparent' },
  commonBtnText: { color: config.colors.themeBlue },
  secondBtnContainer: { marginTop: 15, backgroundColor: config.colors.themeLightBlack },
  thirdBtnContainer: { marginTop: 15 },
  sliderBtnContainer: { marginTop: 15, marginBottom: 30 },
  headerLeftContainer: { alignSelf: 'flex-start', paddingHorizontal: 10 },
  headerLeftText: { color: config.colors.white },
});

export default Home;
