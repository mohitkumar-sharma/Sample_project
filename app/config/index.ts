import colors from './colors';
import localConfig from './localConfig';
import strings from './strings';
import routes from './routes';
import fontWeights from './fontWeights';
import accessibilityStrings from './accessibilityStrings';
import { store } from './store';

const config = {
  colors,
  ...localConfig,
  store,
  strings,
  routes,
  fontWeights,
  accessibilityStrings,
};

export default config;
