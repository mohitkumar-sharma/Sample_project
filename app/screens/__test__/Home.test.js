import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Home from '../Home';
import config from '../../config';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';

describe('Home', () => {
  const dispatch = jest.fn();
  redux.useDispatch.mockReturnValue(dispatch);

  const navigation = {
    navigate: jest.fn(),
    setOptions: jest.fn(),
  };

  /* Mocking store */
  const mockStore = configureStore([]);
  const store = mockStore({ deviceInfo: {}, isShowWarningBanner: true });

  jest
    .spyOn(redux, 'useSelector')
    .mockImplementationOnce(() => store.getState().isShowWarningBanner)
    .mockImplementationOnce(() => store.getState().deviceInfo);

  it('matches snapshot', () => {
    const { toJSON } = render(<Home navigation={navigation} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('pressing buttons', () => {
    const buttonPressed = jest.fn();
    const { getAllByTestId } = render(<Home navigation={navigation} />);

    const buttons = getAllByTestId(config.accessibilityStrings.NORMAL_BUTTON);
    buttons.forEach(async (button) => {
      await fireEvent.press(button);
      expect(buttonPressed).toBeCalledTimes(1);
    });
  });

  it('sliding button slides', async () => {
    const { getByTestId } = render(<Home navigation={navigation} />);

    const buttonContainer = getByTestId(config.accessibilityStrings.SLIDER_BUTTON_CONTAINER);
    const [INVSliderButton] = buttonContainer.children;
    INVSliderButton.props.onEndReached();
  });
});
