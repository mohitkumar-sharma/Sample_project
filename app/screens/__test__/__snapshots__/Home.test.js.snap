// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1A1924",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFCC00",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 15,
            "fontWeight": "400",
            "margin": 10,
            "padding": 3,
          },
          undefined,
        ]
      }
      testID="banner-text"
    >
      Notice: You are running this app on true
    </Text>
  </View>
  <View
    style={
      Object {
        "bottom": 20,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFCC00",
          "fontSize": 18,
          "fontWeight": "500",
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      4 variations of a button
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 8,
          "height": 55,
          "justifyContent": "center",
          "marginHorizontal": 20,
          "marginTop": 15,
          "opacity": 1,
        }
      }
      testID="normal-button"
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "400",
              "textAlign": "center",
            },
            Object {
              "color": "#6EB1F7",
            },
          ]
        }
        testID="normal-button-text"
      >
        Press me
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#34424A",
          "borderRadius": 8,
          "height": 55,
          "justifyContent": "center",
          "marginHorizontal": 20,
          "marginTop": 15,
          "opacity": 1,
        }
      }
      testID="normal-button"
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "400",
              "textAlign": "center",
            },
            Object {
              "color": "#6EB1F7",
            },
          ]
        }
        testID="normal-button-text"
      >
        Press me
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#6EB1F7",
          "borderRadius": 8,
          "height": 55,
          "justifyContent": "center",
          "marginHorizontal": 20,
          "marginTop": 15,
          "opacity": 1,
        }
      }
      testID="normal-button"
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "400",
              "textAlign": "center",
            },
            undefined,
          ]
        }
        testID="normal-button-text"
      >
        Press me
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 30,
          "marginTop": 15,
        }
      }
      testID="slider-button-container"
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "borderColor": "#6EB1F7",
              "borderRadius": 9,
              "borderWidth": 1,
              "height": 55,
              "justifyContent": "center",
              "marginHorizontal": 20,
            },
            Object {
              "alignItems": "flex-start",
            },
          ]
        }
      >
        <View
          onLayout={[Function]}
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#6EB1F7",
                  "borderRadius": 8,
                  "height": 55,
                  "justifyContent": "center",
                  "width": 55,
                },
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 20,
                  },
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 25,
                    },
                    undefined,
                  ],
                  Object {
                    "fontFamily": "simple-line-icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="slider-button-icon"
            >
              ÓÅÉ
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "position": "absolute",
                "zIndex": -1,
              },
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#6EB1F7",
                  "fontSize": 17,
                  "fontWeight": "400",
                },
                Object {
                  "color": "#6EB1F7",
                },
              ]
            }
            testID="slider-button-text"
          >
            Slide me to continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
